### ISSUE TITLE
BUG: TURP ROW Usage Fees are not updated or calculating Properly (On PROD🎉🎉)

### ISSUE THREAD

**Issue by Nadin-Nader** [cityofaustin/atd-data-tech#7673]
TURP ROW Usage Fees are not updated or calculating Properly
In all tiers the following fees were not correct:
Sidewalk Space Extension, Alley extension, Additional traffic lane extension

Link to Fees:
https://www.austintexas.gov/sites/default/files/files/Transportation/Right_of_Way/Fee%20Schedule%202021-2022.pdf

**Comment by Nadin-Nader** [comment]
@Sreelatha-BlazeKannan here is the link to the fee codes that Tracy has. Please let me know if this will work. If not i can try to get it some other way (The fee amounts are in the link in the AC.
https://drive.google.com/drive/u/1/folders/17nc2pJqTVMqo7CncLaPKwWYJgwVpTGSi

**Comment by Sreelatha-BlazeKannan** [comment]
@Nadin-Nader , @TracyLinder - I got the fee codes for the below fees. However, I am not able to get the amount for these. When I saw the lookup table I see the fee amounts as below. Could you please confirm the fee amounts?
Sidewalk Space Extension - 50312
Alley extension - 50357
Additional traffic lane extension - 50320

![image](https://user-images.githubusercontent.com/44907715/140997322-095b0f36-8ed1-46cd-a3d6-b66acff31c48.png)

**Comment by Nadin-Nader** [comment]
@Sreelatha-BlazeKannan Here is what Paloma said:
[2:33 PM] Amayo-Ryan, Paloma


No problem!  I've circled the fee amounts below.  What's tricky about these is that the fees change based on the duration of the permit:

![Fees Not calculating.png](https://images.zenhubusercontent.com/5e8dfbfabf171441c3a5684e/5cc35c7e-18a1-43c2-8d54-bf32ab3bdb1d)

**Comment by Sreelatha-BlazeKannan** [comment]
@Nadin-Nader , @TracyLinder - Submitted DEV Deployment plan; Please approve and create task ID for me for the commit ID: b7265037de81

**Comment by Nadin-Nader** [comment]
**Status:**
Sent to the DevOps CM Team to be deployed to **DEV** under **_SCTASK0179661_**

**Comment by TracyLinder** [comment]
### Please promote to TEST so UAT can begin.

**Comment by Sreelatha-BlazeKannan** [comment]
@Nadin-Nader , @TracyLinder  -  #7673 TEST DP submitted; please approve the commit ID: b7265037de81 and create task id for me.

**Comment by Nadin-Nader** [comment]
@Sreelatha-BlazeKannan this has been sent to the DEVOPS CM Team to be deployed to **_TEST_** under **SCTASK0180233**

**Comment by Nadin-Nader** [comment]
@Sreelatha-BlazeKannan This has been deployed to TEST. Please UAT

**Comment by Nadin-Nader** [comment]
@Sreelatha-BlazeKannan @TracyLinder here is what kim came back with on UAT from TEST:

Testing 7673- TURP ROW Usage Fees
RW-PVT 21-048878 RW (Initial Issuance)
-	Insert & Run Procedure does not insert TURP ROW Usage Fees!
-	After completing TURP Review, no ROW Usage Fees were inserted.
RW-PVT 21-048608 RW (Extension)
-	Tier 3 Alley Extension Fee did not calculate correctly. It should be the same as the 1st traffic lane extension in Tier 3 if the dimensions are the same, which they were. (see image below)
 
![image.png](https://images.zenhubusercontent.com/5e8dfbfabf171441c3a5684e/78b39f53-2089-4206-963e-485ff9a25098)

**Comment by Sreelatha-BlazeKannan** [comment]
@Nadin-Nader , @TracyLinder - Alley extension calculation for tier 3 fee not working. issue fixed and submitted the DEV Deployment plan.  approve and create task id for me. Commit id: 6375721046df

**Comment by Nadin-Nader** [comment]
@Sreelatha-BlazeKannan This has been sent to the DevOps CM Team to be deployed to **_DEV_** under **SCTASK0182899**

**Comment by Nadin-Nader** [comment]
@Sreelatha-BlazeKannan this has been deployed to **_DEV_**, Please UAT and let me know the outcome.

**Comment by Sreelatha-BlazeKannan** [comment]
@Nadin-Nader , @TracyLinder - Not able to test in DEV as I am not having access to add Street & Property information alley extension details.

**Comment by Nadin-Nader** [comment]
@Sreelatha-BlazeKannan this has been sent to the DevOps CM Team to be deployed to **_TEST_** under **SCTASK0183986**

**Comment by Nadin-Nader** [comment]
**Status:**
Waiting on the DevOps CM Team to deploy to **_TEST_** under **SCTASK0183986**

**Comment by Sreelatha-BlazeKannan** [comment]
@Nadin-Nader , @TracyLinder  - It is successfully promoted to TEST. All are working as expected. Alley extension is calculated with new amounts. Ready to Stakeholders UAT and PROD Promotion.

**Comment by Nadin-Nader** [comment]
@Sreelatha-BlazeKannan @TracyLinder this has been approved to go to **_PROD_** per Kim. Her UAT notes are below:

> [Yesterday 5:55 PM] Perez, KimberlyTest Results for Ticket 7673
>     
> TURP ROW Usage fees for extensions calculated correctly. The Alley Tier 3 extension has been fixed. yes
> 
> OBSERVATION:
> The issue of ROW Usage fees not inserting during initial issuance still persists. See ticket 7829.
> Still adding TDS Tech Surcharge. See ticket 7707.

**Comment by Nadin-Nader** [comment]
## Approved for **_PROD_** Deployment 🎉🎉🎉

**Comment by Sreelatha-BlazeKannan** [comment]
@Nadin-Nader , @TracyLinder - Submitted PROD DP. Please create task id for me. Here is the order of commit ID's for PROD Promotion.

Order 1 - Commit ID: b7265037de81
Order 2 – Commit ID: 245111da7435

**Comment by Nadin-Nader** [comment]
@Sreelatha-BlazeKannan This has been sent to the DevOps CM Team to be deployed to _**PROD**_ under **SCTASK0185394**

**Comment by Nadin-Nader** [comment]
**Status:**
Waiting on the DevOps CM Team to deploy this to **_PROD_** under **SCTASK0185394**

**Comment by Nadin-Nader** [comment]
@Sreelatha-BlazeKannan @TracyLinder this has been deployed to PROD. I will send an email to the stakeholders and let them know and close after our 2pm 12/10/21

**Comment by Nadin-Nader** [comment]
Email sent:

> Hi Kim and Paloma,
> 
> Hope you both are doing well. I wanted to inform you that the fix for Card #7673 BUG: TURP ROW Usage Fees are not updated or Calculating Properly in now on PROD. 
> Please verify on PROD and let me know if all is well at our 2pm meeting tomorrow 12/10.
> I will keep the card open and close it if no more issues pertaining to this hare found.
> 
> Respectfully,
> 
> Nadin A. Nader (He/Him/His)
> Product Delivery Manager
> Data and Technology Services
>

**Comment by Nadin-Nader** [comment]
Approved to close

**Comment by Nadin-Nader** [comment]
@TracyLinder this has bugs on PROD

**Comment by SubhaShroff** [comment]
@Nadin-Nader  Sree is working on this ticket. I am tagging her here. 
@Sreelatha-BlazeKannan

**Comment by Nadin-Nader** [comment]
**Status:**
Kim is testing this and will let me know if this is still an issue since it stopped doubling the fees.
New card created for the fees not calculating if Parking parameters (from the Property and Street Info tab) are added: Card #8000

**Comment by Nadin-Nader** [comment]
All is well per Paloma. Closing

**Comment by Nadin-Nader** [comment]
All is well per Paloma. Closing
